docusaurus/docs/React/guides/customization/pin-indicator.mdx

```diff
@@ -1,262 +1,102 @@
+const CustomPinIndicator = () => {
+  const { message } = useMessageContext('CustomPinIndicator');
 
-```tsx
-const CustomPermissions = {
-  customChannelType: {
-    admin: true,
-    anonymous: false,
-    channel_member: false,
-    channel_moderator: true,
-    guest: false,
-    member: false,
-    moderator: true,
-    owner: true,
-    user: false,
-  },
-};
-```
+  const pinnedBy = message.pinned_by?.name || message.pinned_by?.id;
 
-```tsx
-<MessageList pinPermissions={CustomPermissions} />
-```
+  if (!pinnedBy) return null;
 
-## Custom Pin Indicator
-
-To utilize pinned messages, we will create a basic message component as well as a custom pin indicator to display when a message has been pinned. The custom pin indicator component will simply display the text 'pinned'. This is where you could get creative with custom icons and styles.
-
-```tsx
-export const CustomPinIndicator = () => {
-  return (
-    <div>
-      <div className='pin-text'>pinned</div>
-    </div>
-  );
+  return <div className='pin-indicator'>📌 Pinned by {pinnedBy}</div>;
 };
-```
-
-```css
-.pin-text {
-  font-style: italic;
-  color: grey;
-}
-```
-
-This component can now be passed to `Channel` and later pulled from the `ComponentContext` when used.
-
-```tsx
-<Channel PinIndicator={CustomPinIndicator}>{/* children of Channel component */}</Channel>
-```
-
-Now we will build a message component that changes its background from turquoise to yellow when a message is pinned. Our custom pin indicator will also replace the `MessageOptions` component unless the message is currently being hovered.
 
-```tsx
-export const CustomMessage = () => {
-  const [hovering, setHovering] = useState(false);
-
-  const { message } = useMessageContext();
-  const { PinIndicator = DefaultPinIndicator } = useComponentContext();
-
-  const messageWrapperRef = useRef<HTMLDivElement>(null);
-
-  const { pinned } = message;
-
-  return (
-    <div
-      className={pinned ? 'pinned-custom-message-wrapper' : 'custom-message-wrapper'}
-      onMouseEnter={() => setHovering(true)}
-      onMouseLeave={() => setHovering(false)}
-    >
-      <div className='custom-message-wrapper-content'>
-        <div className='custom-message-header'>
-          <div className='custom-message-header-name'>{message.user?.name || message.user?.id}</div>
-        </div>
-        <MessageText />
-        <MessageRepliesCountButton reply_count={message.reply_count} />
-      </div>
-      <div className='custom-message-right-wrapper'>
-        {hovering ? (
-          <MessageOptions messageWrapperRef={messageWrapperRef} />
-        ) : (
-          pinned && <PinIndicator />
-        )}
-      </div>
-    </div>
-  );
+export const ChannelWrapper = ({ children }) => {
+  return <Channel PinIndicator={CustomPinIndicator}>{children}</Channel>;
```

Comment:
I usually add a little explanation of the `ComponentContext` in every cookbook article, otherwise it may seem weird that we're passing the custom component to the `Channel`. You can copy-paste something from the Message Input cookbook if you want to!

Reply:
As per what we talked about yesterday - instead of this `WrappedChannel` and the whole app boilerplate we'll be going with this simplified solution unless agreed upon otherwise. I think it works well enough for our integrators to copy stuff over to their source files as they already have their own wrappers probably. 

```tsx
const CustomComponent = () => {...}

<Channel Component={CustomComponent} >...</Channel>
```